extension/dnsmasq/dnsmasq.js

```diff
@@ -1452,7 +1453,14 @@ module.exports = class DNSMASQ {
   //save data in redis
   //{mac:{ipv4Addr:ipv4Addr,name:name}}
   //host: { ipv4Addr: '192.168.218.160',mac: 'F8:A2:D6:F1:16:53',name: 'LAPTOP-Lenovo' }
-  async setupLocalDeviceDomain(restart, hosts, isInit) {
+  async setupLocalDeviceDomain(hosts, isInit) {
+    if (this.updatingLocalDomain) {
```

Comment:
If there is a scheduled update, no more scheduled update should be added, simply return.

for current code, if setupLocalDeviceDomain is called X times, the file will be updated X times still. If should be updated only once.